src/configuration.rs

```diff
@@ -33,30 +44,47 @@ impl UpdateSection {
 		}
 	}
 
-	pub fn set_has_update(&mut self, has_update: bool) {
-		self.has_update = has_update;
+	pub fn set_update_check_time(&mut self) {
 		self.last_checked = SystemTime::now()
 			.duration_since(UNIX_EPOCH)
 			.unwrap_or_else(|_| Duration::from_secs(0))
 			.as_secs();
 	}
 }
 
-#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+#[derive(Debug, Default, PartialEq, Clone, Serialize)]
 pub struct Configuration {
 	pub window: WindowSection,
 	pub bindings: Option<BTreeMap<String, Vec<String>>>,
 	pub updates: UpdateSection,
 }
 
+#[derive(Deserialize)]
+pub struct IncompleteConfiguration {
+	pub window: Option<WindowSection>,
+	pub bindings: Option<BTreeMap<String, Vec<String>>>,
+	pub updates: Option<UpdateSection>,
+}
```

Comment:
I don't mean to be nitpicking but it's quite suspicous that there are two alomst identical structs. I'm wondering why don't we just make the `UpdateSection` optional i.e. `Option<UpdateSection>`. If it's a `None`, the updater code can simply do the default behaviour which is to check for updates and then set the update section to the approproate `Some`. I peeked into #48 just to see if you did it in a different way there but something similar appears there as well.

Reply:
The `Option`s are needed for forwards compatibility, if new options are added in the future. I convert it to a struct without `Option`s so we don't need `unwrap` everywhere.

Reply:
This seems to me that it comes down to personal preference as this is about having a few extra pattern matches vs having an additionnal struct. I appreciate the time you invested working on this so I'll accept your decision and I'll merge this as it is.